VideoLocker/src/main/java/org/edx/mobile/view/UserProfileFragment.java

```diff
@@ -168,26 +171,25 @@ private void setAccount(@Nullable final Account account) {
                 viewHolder.locationContainer.setVisibility(View.GONE);
             } else {
                 viewHolder.limitedView.setVisibility(View.GONE);
-                if (account.getLanguageProficiencies().isEmpty()) {
-                    viewHolder.languageContainer.setVisibility(View.GONE);
-                } else {
-                    viewHolder.languageContainer.setVisibility(View.VISIBLE);
-                    viewHolder.languageText.setText(
-                            new Locale.Builder()
-                                    .setLanguage(account.getLanguageProficiencies().get(0).getCode())
-                                    .build()
-                                    .getDisplayName());
+                viewHolder.languageContainer.setVisibility(View.GONE);
+                if (!account.getLanguageProficiencies().isEmpty()) {
+                    try {
+                        viewHolder.languageText.setText(
+                                LocaleUtils.getLanguageNameFromCode(account.getLanguageProficiencies().get(0).getCode()));
+                        viewHolder.languageContainer.setVisibility(View.VISIBLE);
+                    } catch (InvalidLocaleException e) {
+                        logger.error(e, true);
```

Comment:
@1zaman Yup, that's exactly what happens. The view is set to GONE by default (line 174) and only set to VISIBLE if we successfully parse the language / country.